### DISCUSSION TITLE
How to plot accumulated slip in a step?

### DISCUSSION THREAD

**Discussion by sushdash** [neperfepx/neper#454] [Q&A] [answered]
Hello!
As per example 4 of FEpX, the syntax to map accumulated slip of a particular slip system is

```
neper -V cycle1-2.sim \
    -simstep 4 \
    -datanodecoo coo \
    -dataeltscaletitle "Accumulated Slip Shear" \
    -dataelt3dcol slip7 \
    -print 4_slipshear7_step4 \
```

But I want to plot it for all the slip systems together, and gave the syntax
```
neper -V cycle1-2.sim \
    -simstep 4 \
    -datanodecoo coo \
    -dataeltscaletitle "Accumulated Slip Shear" \
    -dataelt3dcol slip \
    -print 4_slipshear_step4 \
```

But it didn't give any result.
Please help me out with this.

**Comment by mkasemer** [comment, marked as answer]
@sushdash - this is more of an FEPX question, but I will answer here:

You can use Neper to calculate a new variable based on what you have output (see various examples in the `-S` section of the documentation). However, this is likely quite complex for your given example, and I would not recommend it.

What I think you may be looking for is the equivalent plastic strain, which is a good metric at describing the total amount of slip shear at a given point. You will have to print this value in your simulation, and then use Neper to process it in the normal way. Please see the FEPX documentation for more on printing variables. For this, you will likely have to re-run your simulation requesting the proper output.

**Reply by rquey** [reply]
To be complete, here is the command to compute the accumulated slip:
```
$ neper -S cycle1-2.sim -reselt 'accumulatedslip:abs(slip1)+abs(slip2)+abs(slip3)+abs(slip4)+abs(slip5)+abs(slip6)+abs(slip7)+abs(slip8)+abs(slip9)+abs(slip10)+abs(slip11)+abs(slip12)'

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.4.2-13
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -S loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] cycle1-2.sim -reselt
         accumulatedslip:abs(slip1)+abs(slip2)+abs(slip3)+abs(slip4)+abs(slip5)+abs(slip6)+abs(slip7)+abs(slip8)+abs(slip9)+abs(slip10)+abs(slip11)+abs(slip12)
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   :     > Input files: tess tesr msh config
Info   :     > Restart id       : 0
Info   :     > Restart files    : yes
Info   :     > Node number      : 8737
Info   :     > Element number   : 5641
Info   :     > Elset number     : 50
Info   :     > Partition number : 4
Info   :     > Step number      : 4
Info   :     > Node results     : coo
Info   :     > Elt results      : sliprate defrate-pl-eq slip
Info   : Running post-processing...
Info   :   - Loading inputs...
Info   :     > simulation.tess...
Info   :     > simulation.msh...
Info   :     > simulation.config...
Info   :   - Writing results...
Info   :     [o] Writing directory `cycle1-2.sim/results/elts'...
Info   :       . accumulatedslip .................................. 100%
Info   :     [o] Wrote directory `cycle1-2.sim/results/elts'.
Info   : Elapsed time: 1.063 secs.
========================================================================
```

@sushdash, I have fixed several bugs (which previously lead to Neper crashes); so, use version 4.4.2-13. You'll have to rerun all Neper -S commands, in FEPX's example 4.

**Reply by sushdash** [reply]
Thank you for explaining in detail.

**Reply by sushdash** [reply]
![Screenshot from 2022-05-13 12-14-58](https://user-images.githubusercontent.com/92076141/168226633-4301331c-1e02-42f0-b63b-41a345b74ce5.png)

I have calculated the accumulated slip and am trying to plot the same using visualisation command. But I am unable to do the same. The error is that the intended file is not found. But I have printed the same file in the intended location.

**Reply by rquey** [reply]
@sushdash - Please copy-paste text as text (as described when opening a discussion), not as an image: it does not look good and prevents us from copying-pasting again.

Anyway, what kind of results do you have in `1_uniaxial.sim`? (`neper -S 1_uniaxial.sim`)
